security/pkg/probe/controller.go

```diff
@@ -171,21 +174,30 @@ func (c *LivenessCheckController) checkGrpcServer() error {
 		CredentialType:      pc.GetCredentialType(),
 		RequestedTtlMinutes: probeCheckRequestedTTLMinutes,
 	}
-	_, err = caProtocol.SendCSR(req)
+	resp, err := caProtocol.SendCSR(req)
 
-	// TODO(incfly): remove connectivity error once we always expose istio-ca into dns server.
-	if err != nil && strings.Contains(err.Error(), balancer.ErrTransientFailure.Error()) {
+	if err == nil && resp != nil && resp.IsApproved {
+		if vErr := util.Verify(resp.SignedCert, privKeyBytes, resp.CertChain, rootCertBytes); vErr != nil {
+			err := fmt.Errorf("CSR sign failure: %v", vErr)
+			log.Errora(err)
+			return err
+		}
+		log.Debugf("CSR signing service is healthy.")
+		if c.checkCount%logEveryNChecks == 1 {
+			log.Infof("Health check #%d: CSR signing service is healthy.", c.checkCount)
+		}
 		return nil
 	}
 
-	return err
+	return fmt.Errorf("CSR sign failure: [error: %v], [resp is nil: %v]", err, resp == nil)
```

Comment:
I would suggest to return this first as what the code originally is. This way you can reduce the nesting level.

Reply:
Sure. Improved.